front/app/components/UI/RHFInput/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Input, InputProps } from '@citizenlab/cl2-component-library';
+import Error, { TFieldName } from 'components/UI/Error';
+import { Controller, useFormContext } from 'react-hook-form';
+import { CLError } from 'typings';
+
+export interface Props extends InputProps {
+  name: string;
+}
+
+const RHFInput = ({ name, ...rest }: Props) => {
+  const {
+    formState: { errors },
+    control,
+  } = useFormContext();
+
+  const defaultValue = '';
+
+  const validationError = errors[name]?.message as string | undefined;
+
+  const apiError =
```

Comment:
I find this line a little complicated... it evaluates to a boolean?

Reply:
Yes, essentially how this works is that the error component knows which error to show based on the field name and api errors that are passed to it. So what we do in order to avoid breaking changes is we use RHF's `setError` functionality to capture any API errors on submit and provide them to the component via the form context. From there - we are checking if there are any with this constant and if so - rely on the `Error` component to display the correct one. Agree it's a bit more complicated than what is has to be but needed to make it backwards compatible with our other forms, at least for now.